cedar-drt/fuzz/fuzz_targets/request-validation.rs

```diff
@@ -0,0 +1,112 @@
+     /// generated schema
+     #[serde(skip)]
+     pub schema: Schema,
+         /// generated hierarchy
+    #[serde(skip)]
+    pub hierarchy: Hierarchy,
+
+    /// the requests to try for this hierarchy and policy. We try 8 requests per
+    /// policy/hierarchy
+    #[serde(skip)]
+    pub requests: [ABACRequest; 8],
+ }
+ 
+ /// settings for this fuzz target
+ const SETTINGS: ABACSettings = ABACSettings {
+     match_types: false,
+     enable_extensions: true,
+     max_depth: 7,
+     max_width: 7,
+     enable_additional_attributes: true,
+     enable_like: true,
+     enable_action_groups_and_attrs: true,
+     enable_arbitrary_func_call: true,
+     enable_unknowns: false,
+     enable_action_in_constraints: true,
+     enable_unspecified_apply_spec: true,
+ };
+ 
+ impl<'a> Arbitrary<'a> for FuzzTargetInput {
+     fn arbitrary(u: &mut Unstructured<'a>) -> arbitrary::Result<Self> {
+         let schema: Schema = Schema::arbitrary(SETTINGS.clone(), u)?;
+         let hierarchy = schema.arbitrary_hierarchy(u)?;
+         let requests = [
+            schema.arbitrary_request(&hierarchy, u)?,
+            schema.arbitrary_request(&hierarchy, u)?,
+            schema.arbitrary_request(&hierarchy, u)?,
+            schema.arbitrary_request(&hierarchy, u)?,
+            schema.arbitrary_request(&hierarchy, u)?,
+            schema.arbitrary_request(&hierarchy, u)?,
+            schema.arbitrary_request(&hierarchy, u)?,
+            schema.arbitrary_request(&hierarchy, u)?,
+        ];
+         Ok(Self { schema, hierarchy, requests })
+     }
+ 
+//  todo 
+     fn size_hint(depth: usize) -> (usize, Option<usize>) {
+         arbitrary::size_hint::and_all(&[
+             Schema::arbitrary_size_hint(depth),
+             Schema::arbitrary_request_size_hint(depth),
+             Schema::arbitrary_request_size_hint(depth),
+             Schema::arbitrary_request_size_hint(depth),
+             Schema::arbitrary_request_size_hint(depth),
+             Schema::arbitrary_request_size_hint(depth),
+             Schema::arbitrary_request_size_hint(depth),
+             Schema::arbitrary_request_size_hint(depth),
+             Schema::arbitrary_request_size_hint(depth), 
+         ])
+     }
+ }
+ 
+ // Non-type-directed fuzzing of (strict) validation.
+ fuzz_target!(|input: FuzzTargetInput| {
+     initialize_log();
+     let def_impl = LeanDefinitionalEngine::new();
+ 
+     // generate a schema
+     if let Ok(schema) = ValidatorSchema::try_from(input.schema) {
+         debug!("Schema: {:?}", schema);
+        let requests = input
+        .requests
+        .into_iter()
+        .map(Into::into)
+        .collect::<Vec<_>>();
+        for request in requests.iter().cloned() {
+            debug!("Request: {request}");
+            let (_, total_dur) =
+                time_function(|| run_req_val_test(&def_impl, schema.clone(), request, Extensions::all_available()));
+            info!("{}{}", TOTAL_MSG, total_dur.as_nanos());
```

Comment:
I'd just delete all the `time_function` and `info!` stuff for the first pass. We can add it back later if we decide we want it.

Reply:
Are you sure? It's not that big of a deal and it just matches up with the rest of the targets better.